packages/ui/src/utils/hooks/useCloseByEscKey.ts

```diff
@@ -0,0 +1,32 @@
+import { useKey } from 'react-use'
+
+type CallbackFunction = (e?: KeyboardEvent) => void
+
+// `callback stack` to track callers of useCloseByEscKey
+const _cbStack: CallbackFunction[] = []
+
+function _removeFromStack(cb: CallbackFunction) {
+  const idx = _cbStack.indexOf(cb)
+  if (idx > -1) {
+    _cbStack.splice(idx, 1)
+  }
+}
+
+export const useCloseByEscKey = (cb: CallbackFunction) => {
+  // 'Esc' -> IE/Edge specific value
+  const predicate = (event: KeyboardEvent) => event.key === 'Escape' || event.key === 'Esc'
```

Comment:
Should we move `predicate` out of the render function?
`useKey` uses it as a dependency for the internal `useMemo` - https://github.com/streamich/react-use/blob/master/src/useKey.ts#L24